### PR TITLE
11.0 mig partner email check

### PR DESCRIPTION
**Pull request by denislour**
Migrate `partner_email_check` to version 11.0 from version 10.0.

### CODE REVIEW

**Comment by denislour** [comment]
@Yajo, I included commits from previous branches.

**Review by congdpt — APPROVED** [review]

File: partner_email_check/README.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (hieulucky111) [inline partner_email_check/README.rst:1]: Follow the guideline: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

**Review by hieulucky111 — APPROVED** [review]
Some minor adjustments, LGTM over all.

File: partner_email_check/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2017 Komit <http://komit-consulting.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Email Format Checker',
+    'version': '11.0.1.0.0',
+    'summary': 'Validate email address field',
+    'author': "Komit, Odoo Community Association (OCA)",
+    'website': 'http://komit-consulting.com',
```
Comment (rafaelbn) [inline partner_email_check/__manifest__.py]: Please change URL as described in https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L9

**Review by rafaelbn — COMMENTED** [review]
Let's test this one. Rebuilding for testing

**Comment by angelmoya** [comment]
Also you can remove copyright info on "\_\_init__.py"

After some remarks LGTM

File: partner_email_check/__manifest__.py (hunk 2 above)
Comment (jcdrubay) [reply partner_email_check/__manifest__.py]: Updated link: https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10

File: partner_email_check/__init__.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Copyright 2017 Komit <http://komit-consulting.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (NL66278) [inline partner_email_check/__init__.py]: Use https: on the license lines of all files.

File: partner_email_check/models/res_partner.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    @api.constrains('email')
+    def constrains_email(self):
+        for rec in self.filtered("email"):
+            self.email_check(rec.email)
+
+    @api.model
+    def email_check(self, email):
+        if validate_email(email):
+            return True
+        else:
```
Comment (NL66278) [inline partner_email_check/models/res_partner.py]: No need for the else. If you do not return on the previous line, you can raise an error.

**Comment by NL66278** [comment]
@denislour Please take up the comments, or answer them. Then I will be able to merge the PR.

**Comment by denislour** [comment]
@NL66278 Sr for my delay, I updated this PR.

**Comment by rafaelbn** [comment]
Very nice feature to include in Odoo core in v13 @fgi-odoo

Like: https://github.com/odoo/odoo/tree/12.0/addons/phone_validation

**Pull request merged**